lib/maze/client/appium/bb_devices.rb

```diff
@@ -71,6 +52,45 @@ def get_device(device_group, platform, platform_version, api_key)
             end
           end
 
+          # Get the id of a device group given its name
+          def get_device_group_id(device_group_name)
+            query = {
+              'filter': "displayName_eq_#{device_group_name}"
+            }
+            devices = query_api('device-groups', query)
+            if devices['data'].size != 1
+              $logger.error "Expected exactly one group with name #{device_group_name}, found #{devices.size}"
+              raise "Failed to find a device group named '#{device_group_name}'"
+            end
+            devices['data'][0]['id']
+          end
+
+          def find_device_in_group(device_group_id)
+            path = "device-groups/#{device_group_id}/devices"
+            query = {
+              'filter': "online_eq_true"
+            }
+            all_devices = query_api(path, query)
+
+            $logger.debug "All available devices in group #{device_group_id}: #{JSON.pretty_generate(all_devices)}"
+            filtered_devices = all_devices['data'].reject { |device| device['locked'] }
+            return filtered_devices.size, filtered_devices.first
```

Comment:
If the log & exception were in this method we wouldn't need to return two things here

Reply:
Haha, I actually refactored this prior to review, moving the log out.  I preferred the multiple return value to having to pass in the group name to log (or not pass it in and only be able to log the group id).